float64[] j0
float64[] j1
float64[] j2 
